include/imgproc/dct_hash.h

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+// Standard C++ includes
+#include <iostream>
+#include <string>
+#include <algorithm>
+
+// Standard C includes
+#include <cmath>
+
+//opencv
+#include <opencv2/opencv.hpp>
+
+
+class DCTHash {
+
+    private:
+    static unsigned int countSetBits(unsigned long long int n) { 
+        unsigned int count = 0; 
+        while (n) { 
+            count += n & 1; 
+            n >>= 1; 
+        } 
+        return count; 
+    } 
+
+    public:
+    
+    
+    //! computes the DCT hash 
+    static unsigned long long computeHash(const cv::Mat &in) {
+        
+		unsigned long long hash;
+        cv::Mat dct_mat;
+        cv::dct(in, dct_mat);
+        cv::Rect roi( 0, 0, 8, 8); 
+        cv::Mat low_dct( dct_mat, roi ); 
```

Comment:
How big will ``dct_mat`` be? Why are we only using part of the image?

Reply:
Well, I think the idea is to throw away the high/low frequency components.

Reply:
Ahh I see. Makes sense.